### ISSUE TITLE
Register multiple listeners with x-on

### ISSUE THREAD

**Issue by gwleuverink** [alpinejs/alpine#173]
I recently came across a situation where I had to do something to the effect of:

```html
<button 
    x-on:click="doSomething()" 
    x-on:touchend="doSomething()"
> 
    Do something!
</button>
```

It would be great if we could register multiple listeners with one directive. For example:

```html
<button  x-on:[click, touchend]="doSomething()" > 
    Do something!
</button>
```

I'm not sure syntax like that is preferred or even possible, but I'm sure there are other use-cases where something like this could come in handy. Would you consider adding something like this in the future?

**Comment by gwleuverink** [comment]
Just wrote up a small proof of concept :) Would love to hear your thoughts

**Comment by SimoTod** [comment]
It's a good concept but I'm a bit unsure about modifiers: It seems like it applies the same ones to each event in the list, am i right?

How do we cover
`@click` + `@keypress.escape.window` (something that somebody could use for closing a modal)?

**Comment by TomS-** [comment]
@SimoTod What about a pipe seperator like what is often used in arrays? 

`@click|keypress.escape.window="doSomething()"`

**Comment by gwleuverink** [comment]
That is a good question. Yes, in the proof of concept right now it applies all modifiers to all handlers at once.
 
```html

<!--
Here the `prevent` modifier would be applied to both handlers
-->

<button 
    x-on:[click,touchend].prevent="doSomething()" > 
    Do something
</button>
```

The `keypress` and `escape` modifiers are a little bit of an odd duck since they have special behaviour.

1. The keypress modifier has it's own modifiers only applicable on keypress events
2. The window modifier registers a listener on a different node than the element itself

If you were to combine them you'd end up with something like this: 
Please note these are nonsensical examples, but you get the gist :)
```html

<!--
This would register both the click and keydown events on the window object (not what you'd want)
Furthermore the escape modifier would be applied to the click event. (won't do anything)
-->

<button
    x-on:[click,keypress].escape.window="doSomething()" > 
    Do something
</button>
```

I'm pondering the following approaches:
``` html

<!--
Group listeners together based on intent
-->

<button href="#some-id"  
    x-on:[click,touchend].prevent="doSomething()" 
    x-on:keypress.escape.window="doSomething()"
> 
    Do something
</button>


<!--
Register modifiers on the event level. 
I don't really like this approach since it drastically affects readability. But it is an idea we can iterate on
-->

<button  
    x-on:[click.prevent,keypress.escape.window]="doSomething()" > 
    Do something
</button>

```

I believe readability really is key and prefer to group them together based on applicable modifies. What do you think @SimoTod

**Comment by TomS-** [comment]
Would you think that:
`
<button  
    x-on:[click,touchend].prevent|keypress.escape.window="doSomething()" > 
    Do something
</button>
`
Could work? So you can use `[]` to chain events with similar modifiers and `|` to add a new event with it's own modifiers. Seems like it can keep it tidy and readable.

**Comment by gwleuverink** [comment]
That is an interesting idea @TomS- 

I can try and add it to #175 if we get some more traction on this. What are your thoughts @calebporzio?

**Comment by SimoTod** [comment]
I think that mixing brackets, commas and pipes would result in something a bit confusing and hard to parse.
Since it's mostly syntax sugar, this PR will be accepted only if the syntax is really clear, probably

I personally don't dislike @TomS-  suggestion of concatenating the events
I.e. `@keypress.window.escape|click|touchend` but I think it's the kind of thing where everyone can have an opinion and most of the solutions are equally good.

**Comment by calebporzio** [comment]
Yeah, closing this. I want to keep the API minimal. See my comment on the PR for more info: #175 

Thanks!
